frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -35,6 +40,77 @@ export interface ClusterRequest {
   certificateAuthorityData?: string;
 }
 
+//refreshToken checks if the token is about to expire and refreshes it if so.
+async function refreshToken(token: string | null) {
+  if (!token || TOKEN_REFRESH_INPROGRESS) {
+    return;
+  }
+  // decode token
+  const decodedToken: any = decodeToken(token);
+
+  // return if the token doesn't have an expiry time
+  if (!decodedToken.exp) {
+    return;
+  }
+  // convert expiry seconds to date object
+  const expiry = decodedToken.exp;
+  const now = new Date().valueOf();
+  const expDate = new Date(0);
+  expDate.setUTCSeconds(expiry);
+
+  // calculate time to expiry in minutes
+  const diff = (expDate.valueOf() - now) / 60000;
+  // If the token is not about to expire return
+  if (diff > MIN_LIFESPAN_FOR_TOKEN_REFRESH) {
```

Comment:
From reading the code it's not clear to me what units diff, or MIN_LIFESPAN_FOR_TOKEN_REFRESH are.

Reply:
@illume  converted both properties to seconds and commented as well